app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java

```diff
@@ -254,133 +260,127 @@ public void onPageScrollStateChanged(int state) {
+                                channelNames.add(channelSelectSpinner3.getSelectedItem().toString());
+                                channelNames.add(channelSelectSpinner4.getSelectedItem().toString());
+                                edgesNames.add(edgeSelectSpinner1.getSelectedItem().toString());
+                                edgesNames.add(edgeSelectSpinner2.getSelectedItem().toString());
+                                edgesNames.add(edgeSelectSpinner3.getSelectedItem().toString());
+                                edgesNames.add(edgeSelectSpinner4.getSelectedItem().toString());
+                                break;
+                            default:
+                                channelNames.add(channelSelectSpinner1.getSelectedItem().toString());
+                                edgesNames.add(edgeSelectSpinner1.getSelectedItem().toString());
                         }
 
-                        @Override
-                        public void onNothingSelected() {
-                            Log.i("Nothing selected", "Nothing selected.");
+                        if (channelMode == 1) {
+                            Thread monitor = new Thread(new Runnable() {
+                                @Override
+                                public void run() {
+                                    captureOne = new CaptureOne();
+                                    captureOne.execute(channelNames.get(0));
+                                    synchronized (lock) {
+                                        try {
+                                            lock.wait();
+                                        } catch (InterruptedException e) {
+                                            e.printStackTrace();
+                                        }
+                                    }
+                                }
+                            });
+                            monitor.start();
+                        } else {
+                            Toast.makeText(getContext(), "Yet to be implemented", Toast.LENGTH_SHORT).show();
                         }
-                    };
-                    logicLinesChart.setOnChartValueSelectedListener(listener);
+                        // Setting cursor to display time at highlighted points
+                        listener = new OnChartValueSelectedListener() {
+                            @Override
+                            public void onValueSelected(Entry e, Highlight h) {
+                                double result = Math.round(e.getX() * 100.0) / 100.0;
+                                xCoordinateText.setText("Time:  " + String.valueOf(result) + " mS");
+                                Log.i("Entry selected", e.toString());
+                            }
+
+                            @Override
+                            public void onNothingSelected() {
+                                Log.i("Nothing selected", "Nothing selected.");
```

Comment:
This kind of logs are not necessary. Just leave it with a comment. Log only the value which has any data attached to it which will be useful in debugging in future.